src/rules.c

```diff
@@ -179,7 +179,7 @@ void SchemaRule_Free(SchemaRule *rule) {
   if (rule->filter_exp) {
     ExprAST_Free((RSExpr *)rule->filter_exp);
   }
-  array_free_ex(rule->prefixes, rm_free(*(char **)ptr));
+  array_free_ex(rule->prefixes, sdsfree(*(sds *)ptr));
```

Comment:
`array_free_ex` is extremely confusing. It requires reasoning about non-local variables. In my opinion, (in a separate PR), it should be rewritten to accept a callback, and pass `ptr` to said callback. That way, all reasoning is local to the scope of the enclosing block.

Reply:
Agree. It should at least get an API similar to `array_foreach` where you can name the variable